lib/chef/resource_builder.rb

```diff
@@ -52,14 +52,6 @@ def build(&block)
       resource.source_line = created_at
       resource.declared_type = type
 
-      # If we have a resource like this one, we want to steal its state
-      # This behavior is very counter-intuitive and should be removed.
-      # See CHEF-3694, https://tickets.opscode.com/browse/CHEF-3694
-      # Moved to this location to resolve CHEF-5052, https://tickets.opscode.com/browse/CHEF-5052
-      if prior_resource && Chef::Config[:resource_cloning]
-        resource.load_from(prior_resource)
```

Comment:
maybe?  i don't see it in chefspec or minitest-chef-handler which are the kinds of tools i would have expected to be poking around this deeply in things.

i don't know how i feel about it, because if anyone in the future wants to do that kind of thing, that's the way to to do it -- and if we ever implemented `def dup` we'd probably allocate a new resource and then more or less call that function.